File: clients/booking-system/src/models/userData.js
import AddressData from "@/models/addressData";

class UserData {
  constructor() {
    this.id = ""
    this.title = "";
    this.givenName = "";
    this.familyName = "";
    this.nickname = "";
    this.emailAddress = "";
    this.isFederatedUser = false;
    this.division = undefined;
    this.address = new AddressData();
  }

  buildAuth0AccountInfo(authUser){
    this.id = authUser.sub;
    this.emailAddress = authUser.email;
    this.givenName = authUser.given_name;
    this.familyName = authUser.family_name;
    this.nickname = authUser.nickname;
    this.isFederatedUser = authUser.sub.includes("google");
  }

  buildUserInfo(userData){
    this.title = userData.title;
    this.division = userData.division;
    this.address = userData.address;
  }
}

export default UserData;

File: api/Models/UsersModels/UserData.cs
using BookingSystem.Api.Models.AddressesModels;
using UserLibrary.Enums;

namespace BookingSystem.Api.Models.UsersModels;

public class UserData
{
    public string? Id { get; set; }
    
    public string? Title { get; set; }
    
    public string? GivenName { get; set; }
    
    public string? FamilyName { get; set; }
    
    public string? Nickname { get; set; }
    
    public string? EmailAddress { get; set; }
    
    public bool IsFederatedUser { get; set; }
    
    public Division Division { get; set; }
    
    public AddressData Address { get; set; } = new();
}